public/app/percona/add-instance/components/AddRemoteInstance/FormParts/NodesAgents/NodesAgents.test.tsx

```diff
@@ -0,0 +1,45 @@
+import { render, screen, waitFor } from '@testing-library/react';
+import React from 'react';
+import { Provider } from 'react-redux';
+
+import { configureStore } from 'app/store/configureStore';
+
+import { NodesAgents } from './NodesAgents';
+import { Form } from 'react-final-form';
+import * as NodesReducer from 'app/percona/shared/core/reducers/nodes/nodes.ts'
+import selectEvent from "react-select-event";
+
+const fetchNodesActionActionSpy = jest.spyOn(NodesReducer, 'fetchNodesAction');
+
+jest.mock('app/percona/inventory/Inventory.service');
+
+describe('Nodes Agents:: ', () => {
+
+  let store = configureStore();
+  let formAPI;
+  render(
+    <Provider store={store}>
+      <Form
+        onSubmit={jest.fn()}
+        render={({ form }) => {
+          formAPI=form;
+          return <NodesAgents form={form}/>
+        }}
+      />
+    </Provider>
+  );
+
+  it('should change the   list of agents when changing the nodes', async () => {
```

Comment:
Also this test doesn't test the things it says. Just checks that the value was selected.

Reply:
Also I've reworked a bit the testing file:
- It isn't good to test the internal implementation of react-final-form, it's better to test that the submitted data matches what we want.

```tsx
import { fireEvent, render, screen, waitFor } from '@testing-library/react';
import React from 'react';
import { Form } from 'react-final-form';
import { Provider } from 'react-redux';
import selectEvent from 'react-select-event';

import * as NodesReducer from 'app/percona/shared/core/reducers/nodes/nodes.ts';
import { configureStore } from 'app/store/configureStore';

import { NodesAgents } from './NodesAgents';

const fetchNodesActionActionSpy = jest.spyOn(NodesReducer, 'fetchNodesAction');

jest.mock('app/percona/inventory/Inventory.service');

const submitMock = jest.fn();

describe('Nodes Agents:: ', () => {
  render(
    <Provider store={configureStore()}>
      <Form
        onSubmit={submitMock}
        render={({ handleSubmit, form }) => (
          <form data-testid="node-agents-form" onSubmit={handleSubmit}>
            <NodesAgents form={form} />
          </form>
        )}
      />
    </Provider>
  );

  it('should change the   list of agents when changing the nodes', async () => {
    await waitFor(() => {
      expect(fetchNodesActionActionSpy).toHaveBeenCalled();
    });

    const form = screen.getByTestId('node-agents-form');

    const nodesSelect = screen.getByLabelText('Nodes');
    await selectEvent.select(nodesSelect, 'pmm-server', { container: document.body });

    fireEvent.submit(form);

    expect(submitMock).toHaveBeenCalledWith(
      expect.objectContaining({
        node: expect.objectContaining({
          value: 'pmm-server',
        }),
        pmm_agent_id: expect.objectContaining({
          value: 'pmm-agent',
        }),
      }),
      expect.anything(),
      expect.anything()
    );
  });
});
```

Reply:
@doracretu3pillar it is not resolved, there are still multiple spaces between the words `the` and `list`:

it('should change the`   `list of agents when changing the nodes')

And the test still doesn't do what it says. (currently it checks if the `pmm-server` node selected the correct pmm agent id, but since it should be selected by default anyway, we don't know if the agent list change was correct or not, since there wouldn't be any change as both the current and newly selected node is the same - same list of agents).